src/Equinox.Core/Infrastructure.fs

```diff
@@ -67,3 +67,4 @@ module Seq =
 module Array =
 
     let inline chooseV f xs = [| for item in xs do match f item with ValueSome v -> yield v | ValueNone -> () |]
+    let inline tryExactlyOneV (xs : _ array) = if xs.Length = 0 then ValueNone else ValueSome xs[0]
```

Comment:
(otherwise this should be tryFirstV)
```suggestion
    let inline tryExactlyOneV (xs : _ array) = if xs.Length <> 1 then ValueNone else ValueSome xs[0]
```

Reply:
🤔 We should only ever get one event, but I'm not sure if we want to short circuit and ignore if we ever get more than one. Might as well just take the first in that case. I'll make it `tryFirstV`